### ISSUE TITLE
SLAM generates multiple maps.

### ISSUE THREAD

**Issue by kanpapa** [ROBOTIS-GIT/turtlebot3#899]
ISSUE TEMPLATE ver. 0.4.0

1. Which TurtleBot3 platform do you use?

    - [X] Burger
    - [ ] Waffle
    - [ ] Waffle Pi

2. Which ROS is working with TurtleBot3?

    - [ ] ROS 1 Kinetic Kame
    - [ ] ROS 1 Melodic Morenia
    - [ ] ROS 1 Noetic Ninjemys
    - [ ] ROS 2 Dashing Diademata
    - [ ] ROS 2 Eloquent Elusor
    - [X] ROS 2 Foxy Fitzroy
    - [ ] etc (Please specify your ROS Version here)

3. Which SBC(Single Board Computer) is working on TurtleBot3?

    - [ ] Intel Joule 570x
    - [ ] Raspberry Pi 3B+
    - [X] Raspberry Pi 4
    - [ ] etc (Please specify your SBC here)

4. Which OS you installed on SBC?

    - [ ] Raspbian distributed by ROBOTIS
    - [ ] Ubuntu MATE (16.04/18.04/20.04)
    - [X] Ubuntu preinstalled server (18.04/20.04)
    - [ ] etc (Please specify your OS here)

5. Which OS you installed on Remote PC?

    - [ ] Ubuntu 16.04 LTS (Xenial Xerus)
    - [ ] Ubuntu 18.04 LTS (Bionic Beaver)
    - [X] Ubuntu 20.04 LTS (Focal Fossa)
    - [ ] Windows 10
    - [ ] MAC OS X (Specify version)
    - [ ] etc (Please specify your OS here)

6. Specify the software and firmware version(Can be found from Bringup messages)

    - Software version: [x.x.x]
    - Firmware version: [V220624R1]

7. Specify the commands or instructions to reproduce the issue.

    - In the process of SLAM with keyboard operation.

8. Copy and Paste the error messages on terminal.

    - Error message not found.
    - See attachment for Rviz2 screen.
    ![rviz2_screenshot_22-08-07_124047](https://user-images.githubusercontent.com/3832390/183274911-8acdec2b-7daf-4d18-afa7-2d08f5744667.png)

9. Please describe the issue in detail.

    - When SLAM is performed, the map may suddenly move to another location. As a result, multiple maps are generated.
    - This does not occur with ROS1 Melodic.

**Comment by kanpapa** [comment]
Another pattern.
![rviz2_screenshot2_2022-08-07 135835](https://user-images.githubusercontent.com/3832390/183276040-28fde5b1-e57e-49fb-9e16-6f2aba4ff6bf.png)

**Comment by kanpapa** [comment]
Continuing, three maps were created.
![rviz2_screenshot3_2022-08-07 140522](https://user-images.githubusercontent.com/3832390/183276282-50cb04e4-9744-4045-939c-5a30223d196d.png)

**Comment by ROBOTIS-Ashe** [comment]
@kanpapa 

Hi. Sorry for the late reply.
The reported issue seems to have a similar cause to #880.
I'll take a look at the OpenCR firmware.
thank you

**Comment by kanpapa** [comment]
Thank you for your reply.
I will also include the OpenCR update log since you mentioned a possible firmware issue.
Thank you in advance.

```
ubuntu@ubuntu:~/opencr_update$  ./update.sh $OPENCR_PORT $OPENCR_MODEL.opencr
aarch64
arm
OpenCR Update Start..
opencr_ld_shell ver 1.0.0
opencr_ld_main
[  ] file name          : burger.opencr
[  ] file size          : 132 KB
[  ] fw_name            : burger
[  ] fw_ver             : V220624R1
[OK] Open port          : /dev/ttyACM0
[  ]
[  ] Board Name         : OpenCR R1.0
[  ] Board Ver          : 0x17020800
[  ] Board Rev          : 0x00000000
[OK] flash_erase        : 1.07s
[OK] flash_write        : 1.33s
[OK] CRC Check          : D26544 D26544 , 0.004000 sec
[OK] Download
[OK] jump_to_fw
ubuntu@ubuntu:~/opencr_update$
```

**Comment by kanpapa** [comment]
I checked odm and the value of position y has suddenly increased.

before
![screenshot_2022-08-10 084618](https://user-images.githubusercontent.com/3832390/183782525-274d3b57-c0bf-4f18-973b-a6263a1c8f10.png)

after
![screenshot_2022-08-10_084649](https://user-images.githubusercontent.com/3832390/183782336-842bd216-69fe-419c-a2e5-7852b4e41100.png)

**Comment by kanpapa** [comment]
This seems to be the same issue #821

**Comment by kanpapa** [comment]
I updated the OpenCR firmware to 0.0.3 and now it works fine.
I will try to run turtlebot3 with foxy in this state for a while.

```
ubuntu@ubuntu:~$ wget https://github.com/ROBOTIS-GIT/OpenCR-Binaries/raw/master/turtlebot3/ROS2/0.0.3/opencr_update.tar.bz2
ubuntu@ubuntu:~$ tar -xjf ./opencr_update.tar.bz2
ubuntu@ubuntu:~$ cd opencr_update/
ubuntu@ubuntu:~/opencr_update$ ./update.sh $OPENCR_PORT $OPENCR_MODEL.opencr
aarch64
arm
OpenCR Update Start..
opencr_ld_shell ver 1.0.0
opencr_ld_main 
[  ] file name   	: burger.opencr 
[  ] file size   	: 127 KB
[  ] fw_name     	: burger 
[  ] fw_ver      	: V190829R3 
[OK] Open port   	: /dev/ttyACM0
[  ]
[  ] Board Name  	: OpenCR R1.0
[  ] Board Ver   	: 0x17020800
[  ] Board Rev   	: 0x00000000
[OK] flash_erase 	: 1.06s
[OK] flash_write 	: 1.29s 
[OK] CRC Check   	: CBBCC4 CBBCC4 , 0.004000 sec
[OK] Download 
[OK] jump_to_fw 
ubuntu@ubuntu:~/opencr_update$ 
```

**Comment by rageorgi** [comment]
> I downgraded the OpenCR firmware from the latest to 0.0.3 and it now works properly. I will try to run turtlebot3 with foxy in this state for a while.
> 
> ```
> ubuntu@ubuntu:~$ wget https://github.com/ROBOTIS-GIT/OpenCR-Binaries/raw/master/turtlebot3/ROS2/0.0.3/opencr_update.tar.bz2
> ubuntu@ubuntu:~$ tar -xjf ./opencr_update.tar.bz2
> ubuntu@ubuntu:~$ cd opencr_update/
> ubuntu@ubuntu:~/opencr_update$ ./update.sh $OPENCR_PORT $OPENCR_MODEL.opencr
> aarch64
> arm
> OpenCR Update Start..
> opencr_ld_shell ver 1.0.0
> opencr_ld_main 
> [  ] file name   	: burger.opencr 
> [  ] file size   	: 127 KB
> [  ] fw_name     	: burger 
> [  ] fw_ver      	: V190829R3 
> [OK] Open port   	: /dev/ttyACM0
> [  ]
> [  ] Board Name  	: OpenCR R1.0
> [  ] Board Ver   	: 0x17020800
> [  ] Board Rev   	: 0x00000000
> [OK] flash_erase 	: 1.06s
> [OK] flash_write 	: 1.29s 
> [OK] CRC Check   	: CBBCC4 CBBCC4 , 0.004000 sec
> [OK] Download 
> [OK] jump_to_fw 
> ubuntu@ubuntu:~/opencr_update$ 
> ```

Hi, is SLAM and navigation with NAV2 working properly for you now in Foxy? Also, did you run these commands directly in Foxy?

**Comment by ROBOTIS-Will** [comment]
An updated OpenCR ROS2 firmware V220127R1 (0.2.1) is released.
Please [download and upload the new firmware](https://emanual.robotis.com/docs/en/platform/turtlebot3/opencr_setup/#opencr-setup) and let me know if this doesn't fix the problem.
Thank you very much for your patience!

**Comment by kanpapa** [comment]
The problem has been resolved in the firmware V220127R1 (0.2.1).
Thank you very much.
